app/presenters/document/field_group_presenter.rb

```diff
@@ -97,53 +98,80 @@ def section_field_search_path(val, field, quoted)
     def section_field_subsection(section)
       field_values = section_field_values(section).compact
       field_values = field_values.slice(0, section[:max]) if section[:max].present?
-
       field_values.map do |val|
-        {}.tap do |item|
-          item[:text] = val
-          if !val.nil? && section[:capitalised]
-            item[:text] = val.titleize
-          end
-          if section[:url]
-            item[:url] = field_value(section[:url])
-          elsif section[:search_field]
-            item[:url] = section_field_search_path(val, section[:search_field], section[:quoted])
-          end
+        section_field_subsection_item(section, val)
+      end
+    end
 
-          # text manipulation
-          item[:text] = format_date(item[:text], section[:format_date])
-
-          if section[:overrides] && item[:text] == section[:override_val]
-            section[:overrides].map do |override|
-              if override[:field_title]
-                item[:text] = override[:field_title]
-              end
-              if override[:field_url]
-                item[:url] = override[:field_url]
-              end
-            end
-          end
+    def section_field_subsection_item(section, val)
+      {
+        text: section_field_subsection_item_text(section, val),
+        url: section_field_subsection_item_url(section, val),
+        ga_data: section[:ga_data],
+        extra_info: section_field_subsection_item_extra_info(section, val)
+      }
+    end
 
-          item[:url] = val if linkable_value?(val)
+    def section_field_subsection_item_extra_info(section, val)
+      return nil unless entity_section?(section) && section[:entity_extra].present?
+      entity = entity_for(section, val)
+      return nil unless entity.present?
+      section_nested_hash(section[:entity_extra], entity)
+    end
 
-          if section[:ga_data]
-            item[:ga_data] = section[:ga_data]
-          end
+    def section_field_subsection_item_url(section, val)
+      if section[:url]
+        return field_value(section[:url])
+      end
 
-          # extra entity info
-          if section[:entity_extra].present?
-            possible_entities = entities(section[:entity_name], section[:entity_proxy_field])
-            salient_entity = possible_entities.detect do |entity|
-              [entity_label(entity, section[:entity_name])].flatten.any? { |label| label == val }
-            end
-            unless salient_entity.nil?
-              item[:extra_info] = section_nested_hash(section[:entity_extra], salient_entity)
-            end
-          end
+      if section[:search_field] && !entity_section?(section)
```

Comment:
This conditional and block needs to go below the `if linkable_entity_section?(section)` part 
Also the ` && !entity_section?(section)` should be removed I think.
Otherwise setting `ENABLE_ENTITY_PAGE=1` doesn't keep the links to search queries.